docs/library/machine.Counter.rst

```diff
@@ -0,0 +1,74 @@
+.. currentmodule:: machine
+.. _machine.Counter:
+
+class Counter -- Pulse/Edge counter
+===================================
+
+The Counter class provides hardware support for counting transitions on an input
+pin.
+
+Example usage::
+
+    from machine import Pin, Counter
+
+    inp = Pin(2, Pin.INPUT)
+    counter = Counter(0, pin=inp, rising=Counter.INCREMENT)  # create a Counter on pin inp
+    counter.start()                                          # start counting transitions
+    count = counter.value()                                  # get the current count
+
+.. Note::
+    The Counter class is currently only implemented for the ESP32 port.
+
+Constructor
+-----------
+
+.. class:: Counter(id, *, pin, rising, falling)
+
+    Create a new Counter object with number ``id``. All ports should support
+    at least one pulse counter with id 0, but may support a number of
+    independent counters. ``pin`` is the :ref:`machine.Pin <machine.Pin>`
+    to be monitored. For all ports, one of either the ``rising`` or
+    ``falling`` keyword arguments may be specified with either the
+    ``Counter.INCREMENT`` or ``Counter.DECREMENT`` constant to specify the
+    edge and direction to count.
+
+    Counters begin in a stopped state, with count 0, and need to be started
+    before they will count transitions of the pin.
+
+Methods
+-------
+
+.. method:: Counter.init(*, ...)
+
+    Reinitialize the Counter. See Constructor for keyword arguments.
+
+.. method:: Counter.deinit()
+
+    Deinitialize the Counter.
+
+.. method:: Counter.start()
+
+    Start the counter, i.e., monitor the pin for transitions.
+
+.. method:: Counter.stop()
+
+    Stop the counter, i.e., finish monitoring the pin for transitions. The
+    current counter value will be retained while the counter is stopped.
+
+.. method::  I2S.value([value])
```

Comment:
Most probably that should be: Counter.value([value])

Reply:
Oops! Cut-and-pasto…